packages/store/src/types.js

```diff
@@ -339,14 +339,14 @@
  * key order equivalence class is always at least as precise as the
  * rank order equivalence class. IOW, if `compareKeys(X,Y) === 0` then
  * `compareRank(X,Y) === 0`. But not vice versa. For example, two different
- * remotables are the same rank but incommensurate as keys.
+ * remotables are the same rank but incomparable as keys.
  *
  * A further invariant is if `compareKeys(X,Y) < 0` then
  * `compareRank(X,Y) <= 0`, i.e., if X is smaller than Y in key order, then X
```

Comment:
```suggestion
 * `compareRank(X,Y) < 0`, i.e., if X is smaller than Y in key order, then X
```
as you corrected above.

Reply:
@erights I'm confident of that claim for every passStyleOf value except "tagged", for which rank order is dictated by rank order of `payload` but key order is dictated by algorithms specific to copySet/copyMap/etc. Is it definitively impossible for e.g. compareKeys of two copySets to return -1 when compareRank of the same two copySets returns 0 or even 1? This is diving into details that I'm not yet familiar with, so I'd like explicit confirmation.

Reply:
I have not tested these. But on paper, I'm failing to generate a counter-example. I *claim* I got it right. Some poking around:

```js
const X = Far('X', {});
const Y = Far('Y', {});
const X = Far('Z', {});
const a = makeTagged('copySet', [X, Y]);
const b = makeTagged('copySet', [Y, X]);
compareKeys(a, b); // is 0 because they represent the same set
compareRank(a, b); // is 0 because all remotables are in the same rank

const c = makeTagged('copySet', [Y, Z]);
compareKeys(a, c); // is NaN because neither is a subset of the other
compareRank(a, c); // is 0 because all remotables are in the same rank
```

Reply:
This invariant is the constraint that guided the co-design or key-order, rank-order, full-order, and the encoding of various keys into taggeds. It is why set elements are sorted into descending order by rank.

Reply:
Would this help explain?

rank order | key order
-----------|----------
 -1               | -1 or NaN
0                 | 0 or NaN
1                 | 1 or NaN

Reply:
> Some poking around:

That suggests that an interesting test would be signed zero, e.g. `makeTagged('copySet', [0, -0])` vs. `makeTagged('copySet', [-0, 0])` (or a trickier way to construct objects of similar shape) as input to `compareKeys` and `compareRank`. What is the best way to get an execution environment where I can test such code, given that I already have a build of this repository?

Reply:
I think they both already normalize `-0` to `0`. But it is worth looking.

> What is the best way to get an execution environment where I can test such code, given that I already have a build of this repository?

You mean interactively, rather than writing tests? You could try our repl, or
https://ses-demo.agoric.app/demos/console/
or
https://jsfiddle.net/dx8jc05f/ 
;)

Please let me know what you found convenient.

Reply:
Hmm, neither of those have `makeTagged`. How do I get to the repl?